assets/app/scripts/services/logs/logLinks.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+angular.module('openshiftConsole')
+  .factory('logLinks', [
+    '$anchorScroll',
+    '$document',
+    '$location',
+    '$window',
+    function($anchorScroll, $document, $location, $window) {
+      var doc = $document[0];
+      return {
+        makeDownload: function(obj) {
+          var a = doc.createElement('a');
+          a.href = $window.URL.createObjectURL(new Blob([obj], {type: 'text/plain'}));
+          a.download = 'log.txt';
+          doc.body.appendChild(a);
+          a.click();
+          doc.body.removeChild(a);
+        },
```

Comment:
Here is the 'download plain text file' function.  [Docs for createObjectUrl](https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL).    

Options:
- we just don't use it :smile: 
- I could check prefixes & use it if it works, fallback to noop:
```javascript
var createObjectURL = (window.URL || window.webkitURL || {}).createObjectURL || _.noop;
```

@jwforres @spadgett fire away!